### PR TITLE
Accountability

### PR DESCRIPTION
**Pull request by msabvid**
Creates accountability function for the liquidator to decide whether liquidation will yield a profit.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ repos:
           - "--max-line-length=110"
           - "--max-complexity=18"
           - "--select=B,C,E,F,W,T4,B9"
-          - "--ignore=W503,E203"
+          - "--ignore=W503,E203,E501"
```
Comment (zombie-einstein) [inline .pre-commit-config.yaml:26]: What is this ignoring?

File: agents/liquidation_agent.py [hunk 2]
```diff
@@ -66,8 +74,94 @@ def __init__(
...
+            [self.id_market, liquidation_address],
+        )[0][1:]
+        seized_assets = collateral_assets // 2
+        liquidation_call_event = self.morpho_blue_abi.liquidate.call(
+            env,
+            self.address,
+            self.morpho_blue_address,
+            [self.market_params, liquidation_address, seized_assets, 0, b""],
+        )[1]
+        if liquidation_call_event:
```
Comment (zombie-einstein) [inline agents/liquidation_agent.py:112]: Is this a boolean value? Or are you checking it's not None?

File: agents/liquidation_agent.py [hunk 3]
```diff
@@ -82,15 +176,20 @@ def update(self, rng: np.random.Generator, env) -> List:
             borrowers_data.append((borrower, health_factor))
 
         # filter risky positions
-        risky_positions = filter(lambda x: x[1] < 0.995, borrowers_data)
-
-        # TODO: do accountability to check whether it is profitable to liquidate
-        # taking into account slippage of Uniswap
-        # Adapt from https://github.com/simtopia/verbs-examples/blob/main/agents/liquidation_agent.py
+        risky_positions = filter(lambda x: x[1] < 0.98, borrowers_data)
```
Comment (zombie-einstein) [inline agents/liquidation_agent.py:179]: Is it worth passing this filtering value as a parameter?

File: results/plot_result.ipynb
Comment (zombie-einstein) [inline results/plot_result.ipynb:1]: Does this need to be a notebook? Would rather a script, notebooks are a pain to maintin

File: .pre-commit-config.yaml (hunk 1 above)
Comment (msabvid) [reply .pre-commit-config.yaml:26]: length of line. I added a link in a comment that was longer than the maximum length

File: results/plot_result.ipynb
Comment (msabvid) [reply results/plot_result.ipynb:1]: I committed this one by error.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (zombie-einstein) [reply .pre-commit-config.yaml:26]: Can you ignore the individual line, or break it up? Would rather keep the check.

File: agents/liquidation_agent.py (hunk 2 above)
Comment (msabvid) [reply agents/liquidation_agent.py]: Just removed this if in a new commit.

File: pyproject.toml [hunk 4]
```diff
@@ -28,7 +28,9 @@ classifiers = [
 dependencies = [
   "verbs",
```
Comment (zombie-einstein) [inline pyproject.toml:29]: Can you pin the verbs version, that way we know the examples will always work

File: pyproject.toml [hunk 5]
```diff
@@ -28,7 +28,9 @@ classifiers = [
...
   "matplotlib",
-  "ipython"
+  "ipython",
+  "plotly",
+  "ipykernel"
```
Comment (zombie-einstein) [inline pyproject.toml:33]: Do you need ipykernel?

File: lltv_recommender.py [hunk 6]
```diff
@@ -316,10 +355,21 @@ def run_sim(
...
         key=key,
         block_number=block_number,
+        seed=10,
+        lltv=lltv,
         sigma=sigma,
         n_steps=n_steps,
         n_borrow_agents=n_borrow_agents,
     )
+    records_borrow_agents = [x[1 : (1 + n_borrow_agents)] for x in results]
+    np.save(
```
Comment (zombie-einstein) [inline lltv_recommender.py:369]: How come this is saved?

**Review by zombie-einstein — COMMENTED** [review]
Couple more questions

File: pyproject.toml [hunk 7]
```diff
@@ -26,11 +26,9 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
 dependencies = [
-  "verbs",
+  "verbs==0.1.0",
   "matplotlib",
   "ipython",
```
Comment (zombie-einstein) [inline pyproject.toml:31]: Does it need ipython?

**Review by zombie-einstein — APPROVED** [review]

**Pull request merged**